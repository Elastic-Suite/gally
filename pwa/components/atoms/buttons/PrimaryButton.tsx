import Button from '@mui/material/Button'
import CommonButtonStyle from "~/components/atoms/buttons/CommonButtonStyle";

const PrimaryButton = (props) => {
  return (
    <Button {...props}
      sx={CommonButtonStyle}
      variant="contained"
    >
      {props.children}
    </Button>
  )
}

export default PrimaryButton
