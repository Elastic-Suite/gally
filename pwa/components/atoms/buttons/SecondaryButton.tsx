import { ReactNode } from 'react'
import { ButtonProps } from '@mui/material/Button'
import { styled } from '@mui/system'

import CommonStyledButton from './CommonButtonStyle'

const SecondaryStyledButton = styled(CommonStyledButton)(({ theme }) => ({
  color: theme.palette.colors.secondary['600'],
  background: theme.palette.colors.secondary['100'],
  boxShadow: theme.palette.colors.shadow.secondaryButton.sm,
  '&::before': {
    background: theme.palette.colors.secondary['200'],
    boxShadow: theme.palette.colors.shadow.secondaryButton.md,
  },
  '&:hover': {
    background: theme.palette.colors.secondary['100'],
    boxShadow: theme.palette.colors.shadow.secondaryButton.sm,
  },
  '&& .MuiTouchRipple-child': {
    backgroundColor: theme.palette.colors.secondary['300'],
  },
}))

interface IProps extends ButtonProps {
  children: ReactNode
}

function SecondaryButton(props: IProps) {
  const { children, ...buttonProps } = props

  return (
    <SecondaryStyledButton {...buttonProps} variant="contained">
      {children}
    </SecondaryStyledButton>
  )
}

export default SecondaryButton
