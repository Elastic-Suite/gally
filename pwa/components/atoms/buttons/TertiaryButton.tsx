import { ReactNode } from 'react'
import { ButtonProps } from '@mui/material/Button'
import { styled } from '@mui/system'

import CommonStyledButton from '~/components/atoms/buttons/CommonButtonStyle'

const TertiaryStyledButton = styled(CommonStyledButton)(({ theme }) => ({
  color: theme.palette.colors.neutral['900'],
  background: 'none',
  boxShadow: 'none',
  '&::before': {
    background: theme.palette.colors.neutral['200'],
    boxShadow: 'none',
  },
  '&:hover': {
    background: 'none',
    boxShadow: 'none',
  },
  '&& .MuiTouchRipple-child': {
    backgroundColor: theme.palette.colors.neutral['300'],
  },
}))

interface IProps extends ButtonProps {
  children: ReactNode
}

function TertiaryButton(props: IProps): JSX.Element {
  const { children, ...buttonProps } = props

  return (
    <TertiaryStyledButton {...buttonProps} variant="contained">
      {children}
    </TertiaryStyledButton>
  )
}

export default TertiaryButton
