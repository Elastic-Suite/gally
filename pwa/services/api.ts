import { apiUrl } from '~/constants'

export function fetchApi<T>(
  language: string,
  url: RequestInfo | URL,
  options: RequestInit = {}
): Promise<T> {
  if (process.env.NEXT_PUBLIC_LOCAL) {
    if (!String(url).startsWith('/')) {
      url = `/${url}`
    }
    if (!String(url).endsWith('.json') && !String(url).endsWith('.jsonld')) {
      url = `${url}.json`
    }
    url = `/mocks${url}`
  }
  url = `${apiUrl}${url}`
  return fetch(url, {
    ...options,
    headers: {
      ...options.headers,
      'Accept-Language': language,
    },
  }).then((response) => response.json())
}
