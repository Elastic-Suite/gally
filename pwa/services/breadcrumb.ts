import { IMenuChild } from '~/store'

export function getSlugArray(data: string[] | string): string[] {
  const newBreadCrumbData = [data[0]]
  for (let index = 0; index < data.length - 1; index++) {
    newBreadCrumbData.push(`${newBreadCrumbData[index]}_${data[index + 1]}`)
  }
  return newBreadCrumbData
}

export function findBreadcrumbLabel(
  findIndex: number,
  slug: string[],
  menu: IMenuChild[],
  deepIndex = 0
): string {
  if (typeof menu === 'object') {
    for (const menuChild in menu) {
      if (menu[menuChild].code === slug[deepIndex]) {
        if (findIndex === deepIndex) {
          return menu[menuChild].label
        }
        return findBreadcrumbLabel(
          findIndex,
          slug,
          menu[menuChild].children,
          deepIndex + 1
        )
      }
    }
  }
  return null
}
