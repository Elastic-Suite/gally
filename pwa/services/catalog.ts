import { ICatalog } from '~/types'

export function findDefaultCatalog(catalogsData: ICatalog[]): ICatalog | null {
  const defaultCatalog = catalogsData
    ? catalogsData.filter((catalog) =>
        catalog.localizedCatalogs.find((localizedCtl) => localizedCtl.isDefault)
      )[0]
    : null

  if (defaultCatalog) {
    defaultCatalog.localizedCatalogs = defaultCatalog.localizedCatalogs.filter(
      (localizedCtl) => localizedCtl.isDefault
    )
  }
  return defaultCatalog ? defaultCatalog : null
}
