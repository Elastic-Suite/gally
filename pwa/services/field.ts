import { IElasticSuiteProperty, IField } from '~/types'

function updateProperties(
  properties: IElasticSuiteProperty,
  key: string,
  value: number | boolean
): IElasticSuiteProperty {
  if (key === 'visible') {
    return {
      ...properties,
      visible: value as boolean,
    }
  } else if (key === 'editable') {
    return {
      ...properties,
      editable: value as boolean,
    }
  } else if (key === 'position') {
    return {
      ...properties,
      position: value as number,
    }
  }
  return properties
}

export function updatePropertiesAccordingToPath(
  field: IField,
  path: string
): IField {
  if (path.includes('admin/settings')) {
    path = 'settings_attribute'
  } else {
    path = path.replaceAll('/', '_').replace('_admin_', '')
  }
  if (field.elasticsuite?.context) {
    const [_, newPropertiesvalues] = Object.entries(field.elasticsuite?.context)
      .filter(([key, _]) => key === path)
      .flat()
    if (newPropertiesvalues) {
      let newProperties = field.elasticsuite
      // eslint-disable-next-line no-return-assign
      Object.entries(newPropertiesvalues).forEach(
        ([property, propertyValue]) =>
          (newProperties = updateProperties(
            newProperties,
            property,
            propertyValue
          ))
      )
      return {
        ...field,
        elasticsuite: newProperties,
      }
    }
  }
  return field
}
