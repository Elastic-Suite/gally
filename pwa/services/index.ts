export * from './api'
export * from './authProvider'
export * from './format'
export * from './hydra'
export * from './jsonld'
export * from './local'
export * from './table'
export * from './tests'
