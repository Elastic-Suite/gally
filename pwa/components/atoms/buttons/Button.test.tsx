import PrimaryButton from './PrimaryButton'
import SecondaryButton from './SecondaryButton'
import TertiaryButton from './TertiaryButton'
import { renderWithProviders } from '~/services'

describe('PrimaryButton', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <PrimaryButton disabled={false} endIcon="" size="medium" startIcon="">
        Hello world
      </PrimaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('SecondaryButton', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <SecondaryButton disabled={false} endIcon="" size="medium" startIcon="">
        Hello world
      </SecondaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('TertiaryButton', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <TertiaryButton disabled={false} endIcon="" size="medium" startIcon="">
        Hello world
      </TertiaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('PrimaryButtonDisabledTrue', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <PrimaryButton disabled endIcon="" size="medium" startIcon="">
        Hello world
      </PrimaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('SecondaryButtonDisabledTrue', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <SecondaryButton disabled endIcon="" size="medium" startIcon="">
        Hello world
      </SecondaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('TertiaryButtonDisabledTrue', () => {
  it('match snapshot', () => {
    const { container } = renderWithProviders(
      <TertiaryButton disabled endIcon="" size="medium" startIcon="">
        Hello world
      </TertiaryButton>
    )
    expect(container).toMatchSnapshot()
  })
})
