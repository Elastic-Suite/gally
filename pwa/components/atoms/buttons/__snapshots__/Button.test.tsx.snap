// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-13u4ebv-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    
    Label
    
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`PrimaryButtonDisabledTrue match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-13u4ebv-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    
    Label
    
  </button>
</div>
`;

exports[`SecondaryButton match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-wby1ea-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    
    Label
    
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`SecondaryButtonDisabledTrue match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-wby1ea-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    
    Label
    
  </button>
</div>
`;

exports[`TertiaryButton match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-13yl9aj-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    
    Label
    
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`TertiaryButtonDisabledTrue match snapshot 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-13yl9aj-MuiButtonBase-root-MuiButton-root"
    disabled=""
    tabindex="-1"
    type="button"
  >
    
    Label
    
  </button>
</div>
`;
